### PR TITLE
convert unsigned to signed when returning bindat traces + gain arg

### PR DESCRIPTION
**Pull request by alejoe91**
If the BinDat extractor is initialized with uint dtype, it might cause errors for some sorters.

This converts uint to int and optionally applies a gain to convert to uV

### CODE REVIEW

File: spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py [hunk 1]
```diff
@@ -17,18 +17,21 @@ class BinDatRecordingExtractor(RecordingExtractor):
         {'name': 'dtype', 'type': 'np.dtype', 'title': "The dtype of underlying data"},
         {'name': 'recording_channels', 'type': 'list', 'value':None, 'default':None, 'title': "List of recording channels"},
         {'name': 'frames_first', 'type': 'bool', 'value':True, 'default':True, 'title': "Frames first"},
-        {'name': 'offset', 'type': 'int', 'value':0, 'default':0, 'title': "Offset in binary file"},
-        {'name': 'probe_path', 'type': 'str', 'title': "Path to probe file (csv or prb)"},
+        {'name': 'geom', 'type': 'list', 'title': "list or array with channel locations"},
```
Comment (samuelgarcia) [inline spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py]: Why this here ?

File: spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py [hunk 2]
```diff
@@ -17,18 +17,21 @@ class BinDatRecordingExtractor(RecordingExtractor):
...
+        {'name': 'offset', 'type': 'int', 'value': 0, 'default': 0, 'title': "Offset in binary file"},
```
Comment (samuelgarcia) [inline spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py]: I think we should change offset to offset_in_file to not mixin gain/offset in amplitude and the one of byte start.

File: spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py [hunk 3]
```diff
@@ -61,6 +64,12 @@ def get_traces(self, channel_ids=None, start_frame=None, end_frame=None):
         else:
             channel_ids = [self._channels.index(ch) for ch in channel_ids]
         recordings = self._timeseries[:, start_frame:end_frame][channel_ids, :]
+        if 'int' in self._dtype and self._dtype.startswith('u'):
```
Comment (samuelgarcia) [inline spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py]: if self._dtype.str.startswith('uint'):

File: spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py (hunk 1 above)
Comment (alejoe91) [reply spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py]: this what? geom? You could pass an array with the locations

Comment (samuelgarcia) [reply spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py]: Why this in this PR I mean.

Comment (samuelgarcia) [reply spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py]: geom

Comment (alejoe91) [reply spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py]: it was already there. The gui params was wrong

Comment (colehurwitz) [reply spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py]: I did not include geom in the GUI params on purpose. It figure it is easier to specify which extractors have default locations (not binary) and for the ones that don't, I force them to give a probe file (for now).

**Comment by colehurwitz** [comment]
Could you remove geom from GUI params for now? Currently, I do not want it as part of the GUI since I can make the user potentially load a .prb or .csv file.

File: spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py (hunk 1 above)
Comment (samuelgarcia) [reply spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py]: So lets remove this 'goem' in gui_params for now. No ?
And then we can merge.

Comment (colehurwitz) [reply spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py]: Yes sounds good.

Comment (alejoe91) [reply spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py]: done

**Comment by alejoe91** [comment]
can I merge?

**Comment by samuelgarcia** [comment]
Yep.

**Pull request merged**